src/devices/bus/spectrum/neogs.cpp

```diff
@@ -178,30 +297,46 @@ void neogs_device::map_io(address_map &map)
 	map(0x0001, 0x0001).mirror(0xff00).lr8(
 		NAME([this](offs_t offset)          { return m_command_in; }));
 	map(0x0002, 0x0002).mirror(0xff00).lr8(
-		NAME([this](offs_t offset)          { m_status &= ~0x80; return m_data_in; }));
+		NAME([this](offs_t offset)          { m_status &= ~0x80; LOGSTATUS(" read: 02 & %02X, status: %02X\n", m_data_in, m_status); return m_data_in; }));
 	map(0x0003, 0x0003).mirror(0xff00).lw8(
-		NAME([this](offs_t offset, u8 data) { m_status |= 0x80; m_data_out = data; }));
+		NAME([this](offs_t offset, u8 data) { m_status |= 0x80; LOGSTATUS("write: 03 & %02X, status: %02X\n", data, m_status); m_data_out = data; }));
 	map(0x0004, 0x0004).mirror(0xff00).lr8(
 		NAME([this](offs_t offset)          { return m_status; }));
 	map(0x0005, 0x0005).mirror(0xff00).lrw8(
-		NAME([this](offs_t offset)          { m_status &= ~0x01; return ~0; }),
-		NAME([this](offs_t offset, u8 data) { m_status &= ~0x01; }));
+		NAME([this](offs_t offset)          { m_status &= ~0x01; LOGSTATUS(" read: 05, status: %02X\n", m_status); return ~0; }),
+		NAME([this](offs_t offset, u8 data) { m_status &= ~0x01; LOGSTATUS("write: 05, status: %02X\n", m_status); }));
 	map(0x0006, 0x0009).mirror(0xff00).lw8(
 		NAME([this](offs_t offset, u8 data) { m_vol[offset] = data & 0x3f; }));
 	map(0x0016, 0x0019).mirror(0xff00).lw8(
 		NAME([this](offs_t offset, u8 data) { m_vol[4 + offset] = data & 0x3f; }));
 	map(0x0010, 0x0010).mirror(0xff00).lw8(
 		NAME([this](offs_t offset, u8 data) { m_mpagx = data; update_config(); }));
-	//map(0x000a, 0x000a).mirror(0xff00).noprw();
-	//map(0x000b, 0x000b).mirror(0xff00).noprw();
+	map(0x000a, 0x000a).mirror(0xff00).lrw8(
+		NAME([this](offs_t offset)          { m_status = (m_status & 0x7f) | (m_mpag << 7); return ~0; }),
+		NAME([this](offs_t offset, u8 data) { m_status = (m_status & 0x7f) | (m_mpag << 7); }));
+	map(0x000b, 0x000b).mirror(0xff00).lrw8(
+		NAME([this](offs_t offset)          { m_status = (m_status & 0xfe) | (m_vol[0] >> 5); return ~0; }),
+		NAME([this](offs_t offset, u8 data) { m_status = (m_status & 0xfe) | ((m_vol[0] >> 5) & 1); }));
```

Comment:
There are quite a few read handlers here that aren’t checking for disabled side effects before updating `m_status`.